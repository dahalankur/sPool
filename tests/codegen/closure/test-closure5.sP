int  x = 123

def quack main():
    int i = 0

    println(int_to_string(x))
    
    while (i < 10):
        # repeatedly defining a function in a loop
        def int hi(): return i;
        i = i + 1
        println(int_to_string(hi()))
        println(int_to_string(x))
        x = x + 1
        ;
    return;

main()