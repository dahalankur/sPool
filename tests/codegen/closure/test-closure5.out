123
0
123
1
124
2
125
3
126
4
127
5
128
6
129
7
130
8
131
9
132
